chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1098 @@
+    [(136000000, 174000001), (400000000, 470000001)],
+    [(446000000, 446200001)],
+    [(462550000, 467725001)],
+    [(476425000, 477412501)]
+]
+
+MEMORY_REGIONS = [
+    ('00022000', 2, 32, 'W', 'chans_present', 'Present Chan.'),
+    ('00032000', 2, 32, 'W', 'chans_scanned', 'Scanned Chan.'),
+    ('00042000', 200, 32, 'W', 'memory', 'Memory Chan. 1-200'),
+    ('00052000', 200, 32, 'S', 'memory2', 'Memory Chan. 201-400'),
+    ('00062000', 100, 32, 'S', 'memory3', 'Memory Chan. 401-500'),
+    ('01002000', 1, 32, 'W', 'func_settings', 'Function Menu'),
+    ('02001000', 16, 16, 'W', 'dtmf_list', 'DTMF Encode List'),
+    ('02011000', 1, 16, 'W', 'dtmf_menu', 'DTMF Menu'),
+    ('02031000', 1, 16, 'W', 'dtmf_start', 'DTMF PTT ID Start'),
+    ('02041000', 1, 16, 'W', 'dtmf_end', 'DTMF PTT ID End'),
+    ('02071000', 1, 16, 'W', 'remote_stun', 'DTMF Remote Stun'),
+    ('02081000', 1, 16, 'W', 'remote_kill', 'DTMF Remote Kill'),
+    ('03011000', 1, 16, 'W', 'scan_menu', 'Scan Menu'),
+    ('03021000', 1, 16, 'W', 'startup_display', 'Startup Display'),
+    ('08002000', 1, 32, 'RO', 'immut_1', 'Serial Number'),
+    ('08002001', 1, 32, 'RO', 'immut_2', 'Production Date')
+]
+
+
+def send_cmd(serial, cmd):
+    try:
+        resp = b''
+        serial.write(cmd)
+        serial.flush()
+        timeout = time() + SERIAL_TIMEOUT
+        while time() < timeout:
+            if serial.inWaiting():
+                resp += serial.read()
+    except Exception as e:
+        raise errors.RadioError(f"Error sending to serial {e}")
+    return resp
+
+
+def start_programming(serial):
+    try:
+        resp = send_cmd(serial, b'PROGRAM')
+        if resp != b'\x06':
+            log.error("Enter Programming failed")
+            raise errors.RadioError("ACK resp not received")
+    except Exception as e:
+        raise errors.RadioError(f"Error Entering Program Mode {e}")
+    return True
+
+
+def end_programming(serial):
+    try:
+        rdata = send_cmd(serial, b'END')
+        if rdata != b'\x06':
+            log.error(f'Error {rdata[0]}')
+    except Exception as e:
+        raise errors.RadioError(f"Error {e}")
+
+
+def parse_response(r):
+    if sum(r[:-1]) % 256 != r[-1]:
+        raise errors.RadioError("Data check sum failed")
+    return r[5:-1]
+
+
+def get_ident(serial):
+    try:
+        bs = send_cmd(serial, b'\x02')
+        model = bs[1:8].decode()
+        mode = bs[8]
+        ver = bs[9:13].decode()
+        log.debug(f'Model {model} Ver {ver} Band Mode {mode}')
+        if SUPPORTED_MODEL_VERSIONS.get(model, []):
+            if ver in SUPPORTED_MODEL_VERSIONS[model]:
+                log.info(f'Model {model} Ver: {ver} supported')
+        else:
+            raise errors.RadioError(
+                f'ERROR: Model: {model} Ver: {ver} is not supported!')
+        return bs
+    except Exception as e:
+        raise errors.RadioError(e)
+
+
+def get_bands(mode):
+    idx = int(hex(mode)[2:]) - 1
```

Comment:
This will fail if the value is ever between 9 and 16:
```
>>> int(hex(12)[2:])
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    int(hex(12)[2:])
    ~~~^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'c'
```
Isn't this just a no-op for anything that doesn't crash?